src/main/java/com/insys/trapps/model/Address.java

```diff
@@ -0,0 +1,170 @@
+package com.insys.trapps.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.insys.trapps.controller.BusinessController;
+
+
+/**
+ * {@link Address Entity} for PersonellServices.
+ *
+ * @author  Kris Krishna
+ * @since 1.0.0
+**/
+
+@Entity
+@Table(name = "address")
+public class Address {
+
+	@Id
+	@GeneratedValue
+	private Long addressId;
+	@Column(nullable = false)
+	private String street;
+	@Column(nullable = false)
+	private String city;
+	@Column(nullable = false)
+	private String state;
+	@Column(name = "ZIP_CODE", nullable = false)
+	private String zip;
+
+	
+	@OneToOne(mappedBy="address")
+	@JsonIgnore
+	private Location location;
+
+	public Address() {
+		super();
+	}
+
+	public Address(String street, String city, String state, String zip) {
+		super();
+		this.street = street;
+		this.city = city;
+		this.state = state;
+		this.zip = zip;
+	}
+
+	public Long getAddressId() {
+		return addressId;
+	}
+
+	public void setAddressId(Long addressId) {
+		this.addressId = addressId;
+	}
+
+	public String getStreet() {
+		return street;
+	}
+
+	public void setStreet(String street) {
+		this.street = street;
+	}
+
+	public String getCity() {
+		return city;
+	}
+
+	public void setCity(String city) {
+		this.city = city;
+	}
+
+	public String getState() {
+		return state;
+	}
+
+	public void setState(String state) {
+		this.state = state;
+	}
+
+	public String getZip() {
+		return zip;
+	}
+
+	public void setZip(String zip) {
+		this.zip = zip;
+	}
+
+	public Location getLocation() {
+		return location;
+	}
+
+	public void setLocation(Location location) {
+		this.location = location;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((addressId == null) ? 0 : addressId.hashCode());
+		result = prime * result + ((city == null) ? 0 : city.hashCode());
+		result = prime * result + ((location == null) ? 0 : location.hashCode());
+		result = prime * result + ((state == null) ? 0 : state.hashCode());
+		result = prime * result + ((street == null) ? 0 : street.hashCode());
+		result = prime * result + ((zip == null) ? 0 : zip.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
there is a lot of unnecessary comparisons in this function. I'm thinking we should only compare the address ids